app/models/framework_response.rb

```diff
@@ -44,12 +52,53 @@ def update_with_flags!(value)
     raise ActiveRecord::ReadOnlyRecord, "Can't update framework_responses because person_escort_record is #{person_escort_record.status}"
   end
 
+  def update_flags_and_clear_dependents!
+    update!(framework_flags: build_flags)
+    clear_dependent_values_and_flags!
+  end
+
   def value=(raw_value)
     unless raw_value.blank? || value_type_valid?(raw_value)
       raise FrameworkResponse::ValueTypeError, raw_value
     end
   end
 
+  def self.bulk_update(person_escort_record, response_values_hash)
```

Comment:
This method is getting a bit bloaty, so considering moving this to a service instead. Thoughts?